### ISSUE TITLE
Search page always returns blank, since upgrading from alpha 66 to alpha 72

### ISSUE THREAD

**Issue by sw-yx** [facebook/docusaurus#4644]
## 🐛 Bug Report

We have written this up in our own issue here: https://github.com/temporalio/documentation/pull/357

TLDR, Since we upgraded from alpha 66 to alpha 72, our search page has been broken (returns no results no matter what you type in)

- Working: https://deploy-preview-326--mystifying-fermi-1bc096.netlify.app/search/?q=workflows
- Broken: https://deploy-preview-327--mystifying-fermi-1bc096.netlify.app/search/?q=workflows

### Have you read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

yes

## Suspected cause

Inspecting the requests, we see that the broken (upgraded) search always includes `facets` and `facetFilters`. removing those fields in cURL causes the request to return results again.

however, i cannot figure out how to remove those fields in docusaurus... specifying `searchParameters` as per [this PR](https://github.com/temporalio/documentation/pull/357/files) has not helped.

## To Reproduce

- PR: https://github.com/temporalio/documentation/pull/327/files upgrading from alpha.66 to alpha.72
- Working: https://deploy-preview-326--mystifying-fermi-1bc096.netlify.app/search/?q=workflows
- Broken: https://deploy-preview-327--mystifying-fermi-1bc096.netlify.app/search/?q=workflows

## Expected behavior

search page behavior should not be broken

## Actual Behavior

search page always returns blank

## Your Environment

- Public source code: https://github.com/temporalio/documentation
- Public site url: https://docs.temporal.io
- Docusaurus version used: v2 alpha.72
- Environment name and version (e.g. Chrome 78.0.3904.108, Node.js 10.17.0): n/a
- Operating system and version (desktop or mobile): n/a

## Reproducible Demo

this is the one that is currently broken: https://github.com/temporalio/documentation

**Comment by slorber** [comment]
As answered on the temporal issue I think you have a DocSearch config issue, which prevents your site to use facet filters and contextual search in general. 

I think fixing the config with the DocSearch team will solve the issues and allow to return results.

The search page is a bit legacy and we'll have to refactor it.
As it's not the most widely used page (most users use the search dropdown) it didn't receive enough attention.

There's an existing issue to track that: https://github.com/facebook/docusaurus/issues/3805

But I believe if you just enable contextual search and fix the config it should be good enough for your use-case.

**Comment by sw-yx** [comment]
thanks very much!

**Comment by xian13** [comment]
Hi @slorber sorry for replying to this issue, but I also get the same problem and hasn't figure it out how to solve it

From reading this issue and temporalio/documentation#357
My understanding to solve this problem is updating the docsearch config

My documentation site is behind http basic auth
The popup is working fine I think because not sending any facet filters
But when using search page, it use facet filters by `docusaurus_tag` and `language` and giving 0 result

Here my docsearch config
```json
{
  "index_name": "docusaurus",
  "start_urls": ["https://example.com/"],
  "sitemap_urls": ["https://example.com/sitemap.xml"],
  "sitemap_alternate_links": true,
  "stop_urls": [],
  "selectors": {
    "lvl0": {
      "selector": "(//ul[contains(@class,'menu__list')]//a[contains(@class, 'menu__link menu__link--sublist menu__link--active')]/text() | //nav[contains(@class, 'navbar')]//a[contains(@class, 'navbar__link--active')]/text())[last()]",
      "type": "xpath",
      "global": true,
      "default_value": "Documentation"
    },
    "lvl1": "header h1",
    "lvl2": "article h2",
    "lvl3": "article h3",
    "lvl4": "article h4",
    "lvl5": "article h5, article td:first-child",
    "lvl6": "article h6",
    "text": "article p, article li, article td:last-child"
  },
  "strip_chars": " .,;:#",
  "custom_settings": {
    "separatorsToIndex": "_",
    "attributesForFaceting": ["language", "version", "type", "docusaurus_tag"],
    "attributesToRetrieve": [
      "hierarchy",
      "content",
      "anchor",
      "url",
      "url_without_anchor",
      "type"
    ]
  }
}
```

Is there still something missing from my config to fix search page?

Thank you

**Comment by xian13** [comment]
Sorry, I already find the problem

The algolia config must be deployed first on the website so it will print docsearch meta tag

**Comment by Sh031224** [comment]
> Sorry, I already find the problem
> 
> The algolia config must be deployed first on the website so it will print docsearch meta tag

This works for me. Thanks U.
